AutomatedTesting/Gem/PythonTests/Terrain/EditorScripts/TerrainHeightGradientList_AddRemoveGradientWorks.py

```diff
@@ -0,0 +1,144 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+class HeightTests:
+    single_gradient_height_correct = (
+        "Successfully retrieved height for gradient1.",
+        "Failed to retrieve height for gradient1."
+    )
+    double_gradient_height_correct = (
+        "Successfully retrieved height when two gradients exist.",
+        "Failed to retrieve height when two gradients exist."
+    )
+    triple_gradient_height_correct = (
+        "Successfully retrieved height when three gradients exist.",
+        "Failed to retrieve height when three gradients exist."
+    )
+    terrain_data_changed_call_count_correct = (
+        "OnTerrainDataChanged called expected number of times.",
+        "OnTerrainDataChanged call count incorrect."
+    )
+
+def TerrainHeightGradientList_AddRemoveGradientWorks():
+    """
+    Summary:
+    Test aspects of the TerrainHeightGradientList through the BehaviorContext and the Property Tree.
+    :return: None
+    """
+
+    import os
+    import math as sys_math
+
+    import azlmbr.legacy.general as general
+    import azlmbr.bus as bus
+    import azlmbr.math as math
+    import azlmbr.terrain as terrain
+    import azlmbr.editor as editor
+    import azlmbr.vegetation as vegetation
+    import azlmbr.entity as EntityId
+
+    import editor_python_test_tools.hydra_editor_utils as hydra
+    from editor_python_test_tools.utils import Report
+    from editor_python_test_tools.utils import TestHelper as helper
+    import editor_python_test_tools.pyside_utils as pyside_utils
+    from editor_python_test_tools.editor_entity_utils import EditorEntity
+
+    terrain_changed_call_count = 0
+    expected_terrain_changed_calls = 0
+
+    aabb_component_name = "Axis Aligned Box Shape"
+    gradientlist_component_name = "Terrain Height Gradient List"
+    layerspawner_component_name = "Terrain Layer Spawner"
+
+    gradient_value_path = "Configuration|Value"
+
+    def create_entity_at(entity_name, components_to_add, x, y, z):
```

Comment:
This exists already as a helper in editor_python_test_tools.editor_entity_utils.py.